### ISSUE TITLE
( $( a )* a b) => ( mod zing {} ) macro not expanded properly 

### ISSUE THREAD

**Issue by jrmuizel** [intellij-rust/intellij-rust#3383]
## Problem description
```rust
Given the following code
macro_rules! zap {
     ( $( a )* a b) => ( mod zing {} )
}
zap!(a a b);
```

asking to show the macro expansion should give `mod zing {}` however the macro does not seem to match properly. rustc is able to expand this properly.

**Comment by vlad20012** [comment]
Is it useful somewhere?

**Comment by jrmuizel** [comment]
Nope. It's a made of an example. It's possible that cases like this show up in the wild someplace though.

**Comment by daboross** [comment]
I use this a number of times in [screeps-game-api](https://github.com/daboross/screeps-in-rust-via-wasm/tree/master/screeps-game-api), for what it's worth.

My particular use case is for creating a bunch of modules with similar methods. This project is bindings to a JavaScript library, and each of the modules made with the macro handles a bunch of similar methods for different JavaScript singleton objects.
